optaplanner-examples/src/test/java/org/optaplanner/examples/tsp/persistence/LowEntityCountTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.tsp.persistence;
+
+import org.junit.Test;
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.examples.tsp.app.TspApp;
+import org.optaplanner.examples.tsp.domain.TspSolution;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalStateException;
+
+public class LowEntityCountTest {
+
+    public static final long LOW_ENTITY_SECONDS_TERMINATION = 1;
+    public static final String PATH_TO_NEARBY_SELECTION_XML = "org/optaplanner/examples/lowentitycount/nearbySelection.xml";
+    private TspApp app = new TspApp();
+
+    @Test
+    public void zeroEntity() {
+        Solver<TspSolution> tspSolutionSolver = app.createSolver(LOW_ENTITY_SECONDS_TERMINATION);
+        TspSolution problem = app.loadUnsolvedProblemFromResource("oneDomicileOnly.xml");
+        assertThatIllegalStateException().isThrownBy(() -> tspSolutionSolver.solve(problem)).
+                withMessageContaining("annotated member").withMessageContaining("must not return");
```

Comment:
Run forever and put a @Test(timeout = ...) on it. That timeout should be 10 times more than your local machine's timeout needs because our jenkins machines are slow. I recomend just setting it to 10 minutes (600_000)

Reply:
@ge0ffrey these tests should not require junit timeout, they should finish very quickly. If it runs forever, it must be fixed.

Reply:
I partially agree with Geoffrey that it's nice to have a mechanism in place that prevents the test from hanging forever in case things don't go as expected. The code is expected to fail fast so we might conclude we don't need any termination because solver isn't actually going to start solving. Now, imagine a bug is introduced and fail-fast feature stops working. Since we have no termination and no timeout on the test, test runs forever → maven build doesn't finish → Jenkins job is aborted. That's something that can be easily avoided and also something so rare that I'm not sure we need to account for it :slightly_smiling_face: 

I vote for a single CH phase that is quick and doesn't need termination. I'd like to avoid time based timeouts with high but random (magic) numbers whenever it's possible.

Reply:
+1 to use CH only, for this test it's sufficient. Even without the fail fast feature, the test still shouldn't run forever - having some kind of solver/phase termination configured.